pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -93,6 +93,7 @@ func ApplyPipelineTaskContext(state PipelineRunState, replacements map[string]st
 		if resolvedPipelineRunTask.PipelineTask != nil {
 			pipelineTask := resolvedPipelineRunTask.PipelineTask.DeepCopy()
 			pipelineTask.Params = replaceParamValues(pipelineTask.Params, replacements, nil)
+			pipelineTask.WhenExpressions = pipelineTask.WhenExpressions.ReplaceWhenExpressionsVariables(replacements)
```

Comment:
I'm a bit confused by this line - this doesn't seem specific to finally tasks, im wondering why pipelinetasks with when expressions currently work without this line? (was there a bug? if so, might make sense to separate that out into a separate commit and/or PR if possible)

Reply:
The `ApplyPipelineTaskContext` function replaces the context variables for execution status. It is written more generally, I guess so it could be easily reused later, but it is currently used on finally tasks only in:

https://github.com/tektoncd/pipeline/blob/08177fab0e2f2d2cc8e30e67f018a552338fca26/pkg/reconciler/pipelinerun/pipelinerun.go#L614-L617

so the line added in this PR is to substitute execution status from pipeline tasks to the when expressions in the finally tasks, so there's no bug

Reply:
ahh i see: since it was only used for finally tasks (and since we dont support status in non finally tasks), got it! thanks for explaining :D